platform/darwin/src/MGLMultiPoint.mm

```diff
@@ -24,105 +24,84 @@ - (instancetype)initWithCoordinates:(CLLocationCoordinate2D *)coords count:(NSUI
 
     if (self)
     {
-        [self setupWithCoordinates:coords count:count];
+        NSAssert(count > 0, @"A multipoint must have coordinates");
+        _coordinates = { coords, coords + count };
+        [self computeBounds];
     }
 
     return self;
 }
 
-- (void)setupWithCoordinates:(CLLocationCoordinate2D *)coords count:(NSUInteger)count
-{
-    if (_coordinates) free(_coordinates);
-
-    _count = count;
-    _coordinates = (CLLocationCoordinate2D *)malloc(_count * sizeof(CLLocationCoordinate2D));
-
-    mbgl::LatLngBounds bounds = mbgl::LatLngBounds::empty();
-
-    for (NSUInteger i = 0; i < _count; i++)
-    {
-        _coordinates[i] = coords[i];
-        bounds.extend(mbgl::LatLng(coords[i].latitude, coords[i].longitude));
-    }
-
-    _bounds = MGLCoordinateBoundsFromLatLngBounds(bounds);
-}
-
-- (void)dealloc
-{
-    free(_coordinates);
-}
-
 - (CLLocationCoordinate2D)coordinate
 {
-    assert(_count > 0);
-
-    return CLLocationCoordinate2DMake(_coordinates[0].latitude, _coordinates[0].longitude);
+    NSAssert([self pointCount] > 0, @"A multipoint must have coordinates");
+    return _coordinates.at(0);
 }
 
 - (NSUInteger)pointCount
 {
-    return _count;
+    return _coordinates.size();
 }
 
 + (NS_SET_OF(NSString *) *)keyPathsForValuesAffectingPointCount
 {
     return [NSSet setWithObjects:@"coordinates", nil];
 }
 
+- (CLLocationCoordinate2D *)coordinates
+{
+    return _coordinates.data();
+}
+
 - (void)getCoordinates:(CLLocationCoordinate2D *)coords range:(NSRange)range
 {
-    if (range.location + range.length > _count)
+    if (range.location + range.length > [self pointCount])
     {
         [NSException raise:NSRangeException
                     format:@"Invalid coordinate range %@ extends beyond current coordinate count of %zu",
-                        NSStringFromRange(range), _count];
+                        NSStringFromRange(range), [self pointCount]];
     }
 
-    NSUInteger index = 0;
+    std::copy(_coordinates.begin() + range.location, _coordinates.begin() + NSMaxRange(range), coords);
+}
 
-    for (NSUInteger i = range.location; i < range.location + range.length; i++)
-    {
-        coords[index] = _coordinates[i];
-        index++;
-    }
+- (void)appendCoordinates:(CLLocationCoordinate2D *)coords count:(NSUInteger)count
+{
+    [self willChangeValueForKey:@"coordinates"];
+    _coordinates.insert(_coordinates.end(), *coords);
```

Comment:
This appends only the first coordinate in the passed-in array. To append the entire array:

```objc
_coordinates.insert(_coordinates.end(), coords, coords + count);
```